vault_program/src/initialize_vault_update_state_tracker.rs

```diff
@@ -75,7 +75,22 @@ pub fn process_initialize_vault_update_state_tracker(
         &vault_update_state_tracker_seeds,
     )?;
 
-    let additional_assets_need_unstaking = if vault.additional_assets_need_unstaking() > 0 {
+    let needs_to_recover_from_partial_or_late_update = {
```

Comment:
do you need to look at anything except for last_full_update_was_last_epoch?

Reply:
No, it should be simplified!